recipes/libinterpolate/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import get, copy
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.52.0"
+
+
+class PackageConan(ConanFile):
+    name = "libinterpolate"
+    description = "A C++ interpolation library with a simple interface that supports multiple interpolation methods."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/CD3/libInterpolate"
+    topics = ("math", "spline", "interpolation", "header-only")
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "19.0",
+            "gcc": "7",
+            "clang": "4",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        pass
+
+    def layout(self):
+        basic_layout(self)
+
+    def requirements(self):
+        self.requires("boost/1.80.0", transitive_headers=True)
+        self.requires("eigen/3.3.7", transitive_headers=True)
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(
+            str(self.settings.compiler), False
+        )
+        if (
+            minimum_version
+            and Version(self.settings.compiler.version) < minimum_version
+        ):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        pass
+
+    def package(self):
+        copy(
+            self,
+            pattern="LICENSE.md",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+        )
+        copy(
+            self,
+            pattern="*.hpp",
+            dst=os.path.join(self.package_folder, "include"),
+            src=os.path.join(self.source_folder, "src"),
+        )
+
+    def package_info(self):
```

Comment:
For header only libs, need to clear the package id so that new variants aren't created for each arch/os/compiler/build type/option combination as per https://docs.conan.io/en/latest/howtos/header_only.html

```suggestion
    def package_id(self):
    self.info.clear()

    def package_info(self):
```

Reply:
The `package_id(...)` method is implemented in the recipe, it is just between the `requirements(...)` and `validate(...)` methods, because that is where it was in the package template. I will move it down.

Reply:
Ah, my bad I must have missed it. A move wasn't necessary :)